### PR TITLE
MM-17937 - add tests for executeCommand action /leave, /settings, /collapse

### PR DESCRIPTION
**Pull request by nadalfederer**
#### Summary
add tests for executeCommand action /leave, /settings, /collapse
#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/12849

### CODE REVIEW

**Comment by nadalfederer** [comment]
Peek 2019-12-24 02-02
Is this right behavior that when we leave favorite channel and reenter it. It will be still favorite? Its current behavior. Should I fix it in this PR?

**Comment by jasonblais** [comment]
@nadalfederer Thanks for your contribution!!! I'll queue this for review but heads up that it may take longer to review because of the holidays.

As for your question regarding behaviour of favorite channels after leaving and re-joining them, this is currently working by design - preferences are restored upon leaving and re-joining a channel: https://mattermost.atlassian.net/browse/MM-6070. However, if this is unexpected behaviour, we'd be open to discussing it.

File: actions/command.test.js [hunk 1]
```diff
@@ -5,9 +5,21 @@ import configureStore from 'redux-mock-store';
 
 import thunk from 'redux-thunk';
 
+import {Client4} from 'mattermost-redux/client';
+
+import * as Channels from 'mattermost-redux/selectors/entities/channels';
+
```
Comment (ali-farooq0) [inline actions/command.test.js]: nit: empty line

File: actions/command.test.js [hunk 2]
```diff
@@ -5,9 +5,21 @@ import configureStore from 'redux-mock-store';
...
+import * as Teams from 'mattermost-redux/selectors/entities/teams';
+
+import {ActionTypes, Constants} from 'utils/constants';
+
```
Comment (ali-farooq0) [inline actions/command.test.js]: nit: empty line

File: actions/command.test.js [hunk 3]
```diff
@@ -5,9 +5,21 @@ import configureStore from 'redux-mock-store';
...
 import * as UserAgent from 'utils/user_agent';
 import * as GlobalActions from 'actions/global_actions.jsx';
```
Comment (ali-farooq0) [inline actions/command.test.js]: empty line

File: actions/command.js [hunk 4]
```diff
@@ -86,7 +87,7 @@ export function executeCommand(message, args) {
             break;
         }
         case '/settings':
-            dispatch(openModal({ModalId: ModalIdentifiers.USER_SETTINGS, dialogType: UserSettingsModal}));
+            dispatch(openModal({modalId: ModalIdentifiers.USER_SETTINGS, dialogType: UserSettingsModal}));
```
Comment (ali-farooq0) [inline actions/command.js]: oh boy, this has been broken for a good while :P

Comment (ali-farooq0) [reply actions/command.js]: Looks like there are a bunch of places where we use `ModalId` instead of `modalId`

Comment (nadalfederer) [reply actions/command.js]: yeah, there are some places where this error occurs

**Pull request merged**